### PR TITLE
Splits controllers and actions

### PR DESCRIPTION
**Pull request by dspages**
Divides responsibility for choosing actions and executing actions into controllers and actions for combat.

### CODE REVIEW

File: Assets/Scripts/CombatController.cs [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
     private Tile currentTile;
 
     public bool isTurn = false;
-    protected bool isMoving = false;
+    protected bool isActing = false;
     [SerializeField] private int move = 5;
      private int actionPoints = 0;
-    [SerializeField] private float moveSpeed = 2;
 
     private float unitHalfHeight = 0;
```
Comment (emshuttles) [inline Assets/Scripts/CombatController.cs]: It looks like this variable isn't used anymore, so let's just delete it and line 23.

File: Assets/Scripts/ActionMove.cs [hunk 2]
```diff
@@ -2,17 +2,89 @@
...
+                Vector3 velocity = SetHorizontalVelocity(direction);
+
+                transform.forward = direction;
+                characterController.Move(velocity * Time.deltaTime);
+            }
+            else
+            {
+                // Center of tile reached
+                transform.position = target;
+                if (path.Count == 1)
```
Comment (emshuttles) [inline Assets/Scripts/ActionMove.cs]: This entire code block isn't necessary due to line 101 of CombatController.

File: Assets/Scripts/Tile.cs [hunk 3]
```diff
@@ -57,8 +58,6 @@ public void ClearMovementVariables()
 
     public void FindNeighbors()
     {
-        Reset();
-
         CheckTile(Vector3.forward);
         CheckTile(Vector3.back);
         CheckTile(Vector3.right);
```
Comment (emshuttles) [inline Assets/Scripts/Tile.cs:63]: I missed this in a previous PR, but can we delete "hit" on line 84?

File: Assets/Scripts/Action.cs [hunk 4]
```diff
@@ -4,15 +4,35 @@
 
 public class Action : MonoBehaviour
 {
+
+    protected int spentActionPoints = 0;
+
+    protected bool inProgress = false;
+
+    protected CombatController combatController;
```
Comment (emshuttles) [inline Assets/Scripts/Action.cs]: If this won't be used in ActionMove, let's make this private.

File: Assets/Scripts/Action.cs [hunk 5]
```diff
@@ -4,15 +4,35 @@
...
 
     // Update is called once per frame
     void Update()
     {
         
     }
+
+    protected void BeginAction(Tile targetTile)
+    {
+        spentActionPoints = 0;
```
Comment (emshuttles) [inline Assets/Scripts/Action.cs:29]: Is this resetting spentActionPoints? Or just a repeat of initialization?

File: Assets/Scripts/ActionMove.cs [hunk 6]
```diff
@@ -2,17 +2,89 @@
...
+                path.Pop();
+            }
+        }
+        else
+        {
+            EndAction();
+        }
+    }
+
+    new public void BeginAction(Tile targetTile)
```
Comment (emshuttles) [inline Assets/Scripts/ActionMove.cs:53]: What does "new" do?

File: Assets/Scripts/ActionMove.cs [hunk 7]
```diff
@@ -2,17 +2,89 @@
...
+    {
+        CalculatePath(targetTile);
+        Move();
+        base.BeginAction(targetTile);
+    }
+
+    protected void CalculatePath(Tile targetTile)
```
Comment (emshuttles) [inline Assets/Scripts/ActionMove.cs]: I think this can be private now.

File: Assets/Scripts/CombatController.cs [hunk 8]
```diff
@@ -1,19 +1,17 @@
 ﻿using System.Collections.Generic;
 using UnityEngine;
 
-public class CombatMovement : MonoBehaviour
+public class CombatController : MonoBehaviour
 {
     protected List<Tile> selectableTiles = new List<Tile>();
     private GameObject[] tiles;
```
Comment (emshuttles) [inline Assets/Scripts/CombatController.cs]: I think we can get rid of this and line 22.

File: Assets/Scripts/CombatController.cs [hunk 9]
```diff
@@ -1,19 +1,17 @@
 ﻿using System.Collections.Generic;
 using UnityEngine;
 
-public class CombatMovement : MonoBehaviour
+public class CombatController : MonoBehaviour
 {
     protected List<Tile> selectableTiles = new List<Tile>();
```
Comment (emshuttles) [inline Assets/Scripts/CombatController.cs]: This can be private.

File: Assets/Scripts/PlayerController.cs [hunk 10]
```diff
@@ -13,19 +13,12 @@ void FixedUpdate()
         {
```
Comment (emshuttles) [inline Assets/Scripts/PlayerController.cs:13]: Why do we use FixedUpdate?

**Review by emshuttles — CHANGES_REQUESTED** [review]
You're on fire this week! Thank you for all the work. Please let me know if you want something different from the way I give feedback.

File: Assets/Scripts/ActionMove.cs (hunk 6 above)
Comment (dspages) [reply Assets/Scripts/ActionMove.cs:53]: That's a hint to confirm that it is intended to override the behavior of the base class. I get a compiler warning if I remove it.

File: Assets/Scripts/PlayerController.cs (hunk 10 above)
Comment (dspages) [reply Assets/Scripts/PlayerController.cs:13]: Good point, better to use update here.

**Comment by dspages** [comment]
> You're on fire this week! Thank you for all the work. Please let me know if you want something different from the way I give feedback.

This is good, thanks for the feedback!

**Pull request merged**